### PR TITLE
feat: prediction no longer takes a time series dataset only table

### LINKED ISSUE #837: feat: change that nn takes no time series for predict
### Is your feature request related to a problem?

right now a nn is trained with a time series and also predicted with a time series. But these timeseries need to be the same so we can just take the information from the trained model and keep it for prediction.

### Desired solution

the prediction should just take a table.

### PR DESCRIPTION
**Pull request by Gerhardsa0**
Closes #837

### Summary of Changes

Prediction no longer takes time series dataset.

Should the model really output the same datatype as the fit data (look abstract class InputConversion), for my Use Case I think it is more intuitive to output a Table instead of a TimeSeriesDataset.

### CODE REVIEW

**Comment by Gerhardsa0** [comment]
@lars-reimann allowing Dataset as type for predict would take a deeper impact to the interface, as the table interface and Image the image interface also need to allow dataset as type. Is this wanted?

**Comment by lars-reimann** [comment]
Let's defer that change until later and only allow tables for now.

File: src/safeds/ml/nn/converters/_input_converter_time_series.py [hunk 1]
```diff
@@ -24,9 +24,9 @@ def __init__(
         self._forecast_horizon = 0
         self._first = True
         self._target_name: str = ""
-        self._time_name: str = ""
         self._feature_names: list[str] = []
         self._continuous: bool = False
+        self._extra_names: list[str] = []
```
Comment (Marsmaennchen221) [inline src/safeds/ml/nn/converters/_input_converter_time_series.py:29]: Add the `_extra names` in the `__eq__`, `__hash__` and `__sizeof__` methods

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

**Review by Marsmaennchen221 — APPROVED** [review]

**Pull request merged**